gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/KafkaJobStatusMonitor.java

```diff
@@ -146,6 +151,19 @@ static void addJobStatusToStateStore(org.apache.gobblin.configuration.State jobS
     String storeName = jobStatusStoreName(flowGroup, flowName);
     String tableName = jobStatusTableName(flowExecutionId, jobGroup, jobName);
 
+    List<org.apache.gobblin.configuration.State> states = stateStore.getAll(storeName, tableName);
+    if (states.size() > 0) {
+      String previousStatus = states.get(states.size() - 1).getProp(JobStatusRetriever.EVENT_NAME_FIELD);
+      String currentStatus = jobStatus.getProp(JobStatusRetriever.EVENT_NAME_FIELD);
```

Comment:
Not sure if getProp() returns null in case of missing key or throws an Exception?

Reply:
It returns null.